expr/func_filterseries.go

```diff
@@ -0,0 +1,84 @@
+package expr
+
+import (
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/consolidation"
+)
+
+type FuncFilterSeries struct {
+	in        GraphiteFunc
+	fn        string
+	operator  string
+	threshold float64
+}
+
+func NewFilterSeries() GraphiteFunc {
+	return &FuncFilterSeries{}
+}
+
+func (s *FuncFilterSeries) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgString{key: "func", val: &s.fn, validator: []Validator{IsConsolFunc}},
+		ArgString{key: "operator", val: &s.operator, validator: []Validator{IsOperator}},
+		ArgFloat{key: "threshold", val: &s.threshold},
+	}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncFilterSeries) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncFilterSeries) getOperatorFunc() func(float64) bool {
+	switch s.operator {
+	case "=":
+		return func(val float64) bool {
+			return val == s.threshold
+		}
+
+	case "!=":
+		return func(val float64) bool {
+			return val != s.threshold
+		}
+
+	case ">":
+		return func(val float64) bool {
+			return val > s.threshold
+		}
+	case ">=":
+		return func(val float64) bool {
+			return val >= s.threshold
+		}
+
+	case "<":
+		return func(val float64) bool {
+			return math.IsNaN(val) || val < s.threshold
+		}
+	case "<=":
+		return func(val float64) bool {
+			return math.IsNaN(val) || val <= s.threshold
+		}
+	}
+	return func(val float64) bool { return false } // should never happen
```

Comment:
Should probably return an `error` that `Exec` can propagate.

Reply:
I validate the input earlier, which is where the error would be thrown. This is just for the compiler to not yell at me but still keep the readability.

Reply:
Right, but 2 different functions do the validation vs the lookup. If one changes and the other doesn't this can trigger. An error would make this much easier to catch than bug hunting.